solutions/CF Packmen.mdx

```diff
@@ -0,0 +1,293 @@
+---
+id: cf-packmen
+source: CF
+title: Packmen
+author: Kevin Sheng
+---
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+Binary search on the minimum time it will take for all the Packmen to eat all the pellets.
+For each Packman, we can greedily calculate the maximum number of pellets that it can eat in the given amount of time and then remove those pellets.
+Note that it is optimal for all of the Packmen to eat continuous segments of the food, and all of the segments must be non-intersecting. They must be continuous because if they weren't, a Packman could always eat it while going from one segment to another, thus making it continuous. Also, they must be non-intersecting because if they *were*, two Packmen would target the same pellet, thus resulting in wasted time.
+Each Packman must also eat all of the uneaten food to its left, or else all the food can't be eaten in the given time. Assigning other Packmen to the uneaten pellets would be pointless, because they would be farther away and thus have no way of reaching the uneatable pellet.
+
+There are two main cases we consider for each Packman: One where the Packman first goes to the right then to the left, and another where the Packman first goes to the *left* then the right. We calculate the food eaten in both of these cases and take the maximum. Note that there is also a special case where a Packman doesn't have any pellets to the left, so it can go to the right, eating as many pellets as time allows.
```

Comment:
```suggestion
There are two main cases we consider for each Packman: one where the Packman first goes to the right and then to the left, and another where the Packman first goes to the *left* and then to the right. We calculate the food eaten in both of these cases and take the maximum. Note that there is also a special case where a Packman doesn't have any pellets to the left, so it can go to the right, eating as many pellets as time allows.
```

Reply:
Also, why is the case where there are no remaining pellets to the left a "special case"?

Reply:
Oh yeah, I wasn't really thinking when I wrote that. I probably put it as a special case because I had to implement an if-statement in my own solution to account for that case.